pkg/notify/mattermost.go

```diff
@@ -0,0 +1,142 @@
+package notify
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/infracloudio/botkube/pkg/config"
+	"github.com/infracloudio/botkube/pkg/events"
+	log "github.com/infracloudio/botkube/pkg/logging"
+	"github.com/mattermost/mattermost-server/model"
+)
+
+// Mattermost contains server URL and token
+type Mattermost struct {
+	Client      *model.Client4
+	Channel     string
+	ClusterName string
+}
+
+// NewMattermost returns new Mattermost object
+func NewMattermost(c *config.Config) Notifier {
+	// Load configurations
+	c, err := config.New()
+	if err != nil {
+		log.Logger.Fatal(fmt.Sprintf("Error in loading configuration. Error:%s", err.Error()))
+	}
+
+	// Set configurations for Mattermost server
+	client := model.NewAPIv4Client(c.Communications.Mattermost.URL)
+	client.SetOAuthToken(c.Communications.Mattermost.Token)
+	botTeam, resp := client.GetTeamByName(c.Communications.Mattermost.Team, "")
+	if resp.Error != nil {
+		log.Logger.Fatal("Error in connecting to Mattermost team. Error: ", resp.Error)
+	}
+	botChannel, resp := client.GetChannelByName(c.Communications.Mattermost.Channel, botTeam.Id, "")
+	if resp.Error != nil {
+		log.Logger.Fatal("Error in connecting to Mattermost channel. Error: ", resp.Error)
+	}
+
+	return &Mattermost{
+		Client:      client,
+		Channel:     botChannel.Id,
+		ClusterName: c.Settings.ClusterName,
+	}
+}
+
+// SendEvent sends event notification to Mattermost
+func (m *Mattermost) SendEvent(event events.Event) error {
+	log.Logger.Info(fmt.Sprintf(">> Sending to Mattermost: %+v", event))
+
+	fields := []*model.SlackAttachmentField{
```

Comment:
SlackAttachmentField should be renamed ?

Reply:
That's a structure in mattermost-server/model.